invoices_app/views.py

```diff
@@ -386,6 +387,40 @@ def approve_invoice(request, pk):
     )
 
 
+@permission_required_decorator(CAN_CHANGE_INVOICE_STATUS_PERM, raise_exception=True)
+def request_changes_invoice(request, pk):
+    status = request.POST.get('status')
+    message = request.POST.get('message')
+    invoice = get_object_or_404(Invoice, pk=pk)
+    if status == invoice_status_lookup(INVOICE_STATUS_CHANGES_REQUEST) and not message:
+        messages.error(request, NO_COMMENT_ERROR)
+        return redirect(
+            reverse(
+                'invoices-detail',
+                kwargs={
+                    'taxpayer_id': invoice.taxpayer.id,
+                    'pk': pk,
+                }
+            )
+        )
+
+    invoice.status = status
+    invoice.changeReason = message
+    invoice.save()
+
+    _send_email_when_change_invoice_status(request, invoice)
+
+    return redirect(
+        reverse(
+            'invoices-detail',
+            kwargs={
+                'taxpayer_id': invoice.taxpayer.id,
+                'pk': pk,
+            }
+        )
+    )
+
+
```

Comment:
This logic is repeated in approve_invoice, we should consider to refactor and have all the repeated logic in one function

Reply:
Done